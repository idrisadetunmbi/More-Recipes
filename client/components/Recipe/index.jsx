import React from 'react';
import PropTypes from 'prop-types';
import './index.scss';

const RecipeCard = (props) => {
  const { recipe, gridStyle } = props;
  return (
    <div id="recipe-card-component" className={`col ${gridStyle}`}>
      <div className="card">
        <div className="card-image">
          <img src={`${recipe.images[0]}`} alt={recipe.title} />
        </div>
        <div className="card-content">
          <span className="card-title">{recipe.title}</span>
          <p>{recipe.description}</p>
        </div>
        <div className="card-action">
          <span><i className="material-icons">thumb_up</i>{recipe.upvotes}</span>
          <span><i className="material-icons">thumb_down</i>{recipe.downvotes}</span>
          <span><i className="material-icons">favorite</i>{recipe.favorites}</span>
        </div>
      </div>
    </div>
  );
};

RecipeCard.propTypes = {
  recipe: PropTypes.shape().isRequired,
  gridStyle: PropTypes.string.isRequired,
};

export default RecipeCard;
