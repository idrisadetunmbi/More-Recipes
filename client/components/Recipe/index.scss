#recipe-card-component {
  .card-image img {
    height: 250px;
    padding: 0.5rem;
  }
  
  .card-title {
    font-size: 19px;
    font-weight: 500;
    display: block;
    display: -webkit-box;
    margin: 0 auto;
    -webkit-line-clamp: 1;
    -webkit-box-orient: vertical;
    overflow: hidden;
    text-overflow: ellipsis;
    height: 2rem;
  }

  .card-content {
    height: 11rem;
  }
  
  .card-content p {
    display: block;
    display: -webkit-box;
    margin: 0 auto;
    -webkit-line-clamp: 3;
    -webkit-box-orient: vertical;
    overflow: hidden;
    text-overflow: ellipsis;
    line-height: 2rem;
  }
  
  .card-action span {
    margin-right: 1.2rem;
    color: #444;
  }
  
  .card-action span i {
    vertical-align: bottom;
    margin-right: 0.3rem;
    font-size: 21px;
  }
}