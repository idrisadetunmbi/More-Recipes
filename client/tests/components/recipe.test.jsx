import React from 'react';
import renderer from 'react-test-renderer';

import Recipe from '../../components/Recipe';

describe('Recipe component', () => {
  test('matches snapshot', () => {
    const props = { recipe: { images: [] } };
    const component = renderer.create(<Recipe {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
