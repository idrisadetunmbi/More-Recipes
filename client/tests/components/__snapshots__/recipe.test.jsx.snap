// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe component matches snapshot 1`] = `
<div
  className="col undefined"
  id="recipe-card-component"
>
  <div
    className="card"
  >
    <div
      className="card-image"
    >
      <img
        alt={undefined}
        src="undefined"
      />
    </div>
    <div
      className="card-content"
    >
      <span
        className="card-title"
      />
      <p />
    </div>
    <div
      className="card-action"
    >
      <span>
        <i
          className="material-icons"
        >
          thumb_up
        </i>
      </span>
      <span>
        <i
          className="material-icons"
        >
          thumb_down
        </i>
      </span>
      <span>
        <i
          className="material-icons"
        >
          favorite
        </i>
      </span>
    </div>
  </div>
</div>
`;
